src/snowflake/connector/auth/oauth.py

```diff
@@ -22,7 +22,7 @@ def type_(self) -> AuthType:
         return AuthType.OAUTH
 
     @property
-    def assertion_content(self) -> str:
+    def assertion_content(self) -> str | None:
```

Comment:
Why added None here

Reply:
The type-hint was incorrect. It doesn't actually affect correctness, but you can see a few lines below that the function `reset_secrets` sets what this returns to `None`